### ISSUE TITLE
j2cli is no longer maintained

### ISSUE THREAD

**Issue by erijo** [yadm-dev/yadm#503]
### Describe the bug

The j2cli tool is no longer maintained (and doesn't work with python 3.12+). The [repository](https://github.com/kolypto/j2cli) has been archived by the owner on Jun 24, 2024. Should be replaced by an alternative, e.g. [jinjanator](https://github.com/kpfleming/jinjanator) or [jinja2-cli](https://github.com/mattrobenolt/jinja2-cli).

[envtpl](https://github.com/envtpl/envtpl) on the other hand seems to be back in business (ping #159).

**Comment by pbsds** [comment]
any progress?

**Comment by boydkelly** [comment]
Any progress?   I dont use this so it doesnt bite.   But this kind of thing raises questions...

**Comment by AndydeCleyre** [comment]
For its popularity, sticking with a jinja tool makes sense. But I want to mention anyway that wheezy.template is an underappreciated great template engine, available via PyPI.

**Comment by erijo** [comment]
> Any progress? I dont use this so it doesnt bite. But this kind of thing raises questions...

Last time I checked I think the problem with the alternatives were that they weren't packaged for Debian. j2cli seems to still be available so it's easy to install.

> But I want to mention anyway that wheezy.template is an underappreciated great template engine, available via PyPI.

But does it have a cli command? For our use-case here we need to be able to call it from the command line in an easy way.

**Comment by AndydeCleyre** [comment]
> But does it have a cli command? For our use-case here we need to be able to call it from the command line in an easy way.

Yes, that's the way I use it. Adapting the example from j2cli's readme:

<details>
<summary>nginx.json:</summary>

```json
{
  "nginx": {
    "hostname": "localhost",
    "webroot": "/var/www/project"
  }
}
```

</details>

<details>
<summary>nginx.wz:</summary>

```nginx
@require(nginx)
server {
  listen 80;
  server_name @nginx['hostname'];

  root @nginx['webroot'];
  index index.htm;
}
```

</details>

```console
$ wheezy.template nginx.wz nginx.json
```

```nginx
server {
  listen 80;
  server_name localhost;

  root /var/www/project;
  index index.htm;
}
```
